Add sections to the consumer concurrency configuration section (#14612)
cloudflare/cloudflare-docs: content/queues/configuration/consumer-concurrency.md

If you have a workflow that is limited by an upstream API and/or system, you may prefer for your backlog to grow, trading off increased overall latency in order to avoid overwhelming an upstream system.

> You can configure the concurrency of your consumer Worker 2 ways:

Reviewer: ```suggestion
To configure the concurrency of your consumer Worker, choose whether to:
```

Author: Outdated since fixed as per Matt's comments

> You can configure the concurrency of your consumer Worker 2 ways:

Reviewer 2: ```suggestion
You can configure the concurrency of your consumer Worker in two ways:
```

> * Set concurrency settings in the Cloudflare dashboard

Reviewer 2: ```suggestion
1. Set concurrency settings in the Cloudflare dashboard
```

If you say a number, use a numbered list.

> * Set concurrency settings via `wrangler.toml`

Reviewer 2: ```suggestion
2. Set concurrency settings via `wrangler.toml`
```

> ### Set concurrency settings in the Cloudflare dashboard:

Reviewer 2: ```suggestion
### Dashboard
```

Author: 1. For the sake of consistency, if we change this title to Dashboard, we may want to consider changing the sister title `### Set concurrency settings via `wrangler.toml`` to ``wrangler.toml``. 
2. I believe that the full title makes it easier to grasp and identify what section to read when looking at the docs (see attached pic)

![image](https://github.com/cloudflare/cloudflare-docs/assets/35609369/55cdbcea-ed1c-44a4-8636-30d085d0e47e)

Reviewer 2: That's fine

> ### Set concurrency settings in the Cloudflare dashboard:

Author: ```suggestion
### Set concurrency settings in the Cloudflare dashboard
```
